client/disputes/test/index.js

```diff
@@ -24,53 +82,137 @@ describe( 'Disputes list', () => {
+			useDisputes.mockReturnValue( {
+				disputes: [],
+				isLoading: false,
+			} );
+
+			const { queryByRole } = render( <DisputesList /> );
+			const button = queryByRole( 'button', { name: 'Download' } );
+
+			expect( button ).toBeNull();
+		} );
+	} );
+
+	describe( 'CSV download', () => {
+		beforeEach( () => {
+			useDisputes.mockReturnValue( {
+				disputes: mockDisputes,
+				isLoading: false,
+			} );
+		} );
+
+		afterEach( () => {
+			jest.resetAllMocks();
+		} );
+
+		afterAll( () => {
+			jest.restoreAllMocks();
+		} );
+
+		test( 'should render expected columns in CSV when the download button is clicked ', () => {
+			const { getByRole } = render( <DisputesList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const expected = [
+				'"Dispute Id"',
+				'Amount',
+				'Status',
+				'Reason',
+				'Source',
+				'"Order #"',
+				'Customer',
+				'Email',
+				'Country',
+				'"Disputed on"',
+				'"Respond by"',
+			];
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvHeaderRow = csvContent.split( os.EOL )[ 0 ].split( ',' );
+			expect( csvHeaderRow ).toEqual( expected );
+		} );
+
+		test( 'should match the visible rows', () => {
+			const { getByRole, getAllByRole } = render( <DisputesList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			const csvContent = downloadCSVFile.mock.calls[ 0 ][ 1 ];
+			const csvRows = csvContent.split( os.EOL );
+			const displayRows = getAllByRole( 'row' );
+
+			expect( csvRows.length ).toEqual( displayRows.length );
+
+			const csvFirstDispute = csvRows[ 1 ].split( ',' );
+			const displayFirstDispute = Array.from(
+				displayRows[ 1 ].querySelectorAll( 'td' )
+			).map( ( td ) => td.textContent );
+
+			// Note:
+			//
+			// 1. CSV and display indexes are off by 1 because the first field in CSV is dispute id,
+			//    which is missing in display.
+			//
+			// 2. The indexOf check in amount's expect is because the amount in CSV may not contain
+			//    trailing zeros as in the display amount.
+			//
+
+			expect(
+				getUnformattedAmount( displayFirstDispute[ 0 ] ).indexOf(
+					csvFirstDispute[ 1 ]
+				)
+			).not.toBe( -1 ); // amount
+
+			expect( csvFirstDispute[ 2 ] ).toBe(
+				`"${ displayFirstDispute[ 1 ] }"`
+			); //status
+
+			expect( csvFirstDispute[ 3 ] ).toBe( displayFirstDispute[ 2 ] ); // reason
+
+			expect( csvFirstDispute[ 5 ] ).toBe( displayFirstDispute[ 4 ] ); // order
+
+			expect( csvFirstDispute[ 6 ] ).toBe(
+				`"${ displayFirstDispute[ 5 ] }"`
+			); // customer
+
+			expect( formatDate( csvFirstDispute[ 9 ], 'Y-m-d' ) ).toBe(
+				formatDate( displayFirstDispute[ 6 ], 'Y-m-d' )
+			); // date disputed on
+
+			expect( formatDate( csvFirstDispute[ 10 ], 'Y-m-d / g:iA' ) ).toBe(
+				formatDate( displayFirstDispute[ 7 ], 'Y-m-d / g:iA' )
+			); // date respond by
+		} );
+	} );
 } );
+
+function getUnformattedAmount( formattedAmount ) {
```

Comment:
As this function and `formatDate` are now used in both Disputes and Deposits tests, does it make sense to move them to a common test utils file?

Reply:
I refactored the tests, this utility functions are not used anymore. I am using `dateI18n` function directly. I like it this way because is closer to the implementation, let me know if you prefer other way.

Reply:
Sounds good 👍